src/admin.rs

```diff
@@ -106,312 +101,549 @@ impl AdminScClient {
     ///     replication: integer   
     ///     rack:    bool
     ///   }
-    /// 
+    ///
     /// or assigned which is vector
     /// [
     ///     [id, [rep1, rep2, rep3]]   
     /// ]
     ///
     #[node_bindgen]
-    async fn create_topic(&self,topic: String, replica: ReplicaParam) -> Result<String,ClientError>  {
+    async fn create_topic(
+        &self,
+        topic: String,
+        spec: TopicSpecWrapper,
+    ) -> Result<String, FluvioError> {
+        if let Some(client) = self.inner.clone() {
+            let mut client = client.write().await.admin().await;
+            client.create(topic.clone(), false, spec.0).await?;
+            Ok(topic)
+        } else {
+            Err(FluvioError::Other("failed to lock client".to_owned()))
+        }
+    }
 
-        let client = self.inner.as_ref().unwrap().clone();
-        let mut client_w = client.write().await;
+    #[node_bindgen]
+    async fn delete_topic(&self, topic: String) -> Result<String, FluvioError> {
+        if let Some(client) = self.inner.clone() {
+            let mut client = client.write().await.admin().await;
+            client.delete::<TopicSpec, String>(topic.clone()).await?;
+            Ok(topic)
+        } else {
+            Err(FluvioError::Other("failed to lock client".to_owned()))
+        }
+    }
 
-        client_w.create_topic(topic,replica.0,false).await
+    #[node_bindgen]
+    async fn list_spu(&self) -> Result<ArrayBuffer, FluvioError> {
+        if let Some(client) = self.inner.clone() {
+            let mut client = client.write().await.admin().await;
+            let spus = client.list::<SpuSpec, _>(vec![]).await?;
+            let json_slice = serde_json::to_vec(&spus).map_err(|err| {
+                FluvioError::Other(format!("serialization error: {}", err.to_string()))
+            })?;
+            Ok(ArrayBuffer::new(json_slice))
+        } else {
+            Err(FluvioError::Other("failed to lock client".to_owned()))
+        }
+    }
 
+    #[node_bindgen]
+    async fn find_partition(&self, topic: String) -> Result<PartitionMetadataWrapper, FluvioError> {
```

Comment:
do we need this method?  this can be built on top of `list_partition`

Reply:
Currently, they return different values. One returns an ArrayBuffer and the other a wrapped value; In the future I'd like to ensure all values are structured data; In a future PR, we can simply.

Reply:
k